lib/sqlalchemy/dialects/mysql/base.py

```diff
@@ -1494,9 +1495,28 @@ def visit_join(self, join, asfrom=False, from_linter=None, **kwargs):
 
     def for_update_clause(self, select, **kw):
         if select._for_update_arg.read:
-            return " LOCK IN SHARE MODE"
+            tmp = " LOCK IN SHARE MODE"
         else:
-            return " FOR UPDATE"
+            tmp = " FOR UPDATE"
+
+        if select._for_update_arg.of and self.dialect._is_mysql:
```

Comment:
**mike bayer** (zzzeek) wrote:

this code checks for "mysql" so that if someone specifies "of" or "skip locked", the code will still invoke successfully on MariaDB which we assume here does not support these features.  

however, if the same code is run on a MySQL version prior to 8, now it fails and will be a regression if current applications are using these flags.   

why allow silent "success" on mariadb but explicit raise on MySQL < 8 ?   this seems inconsistent to me.   this likely should be testing MySQL server version as well and especially if we want this in 1.3.x.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/1928

Reply:
**Gord Thompson** (gordthompson) wrote:

Will work on testing MySQL server version

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/1928

Reply:
**Gord Thompson** (gordthompson) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/1928